### PR TITLE
Single AMI changes

### PR DESCRIPTION
**Pull request by webvictim**
- Add single AMI builds - we now have one OSS and one Enterprise AMI built which will be used for both AWS Marketplace and Terraform
- Add support for Amazon Certificate Manager to both AWS Marketplace products and Terraform
- Add support for Letsencrypt to AWS Marketplace products
- Move/rename old Cloudformation example code

### CODE REVIEW

**Comment by klizhentas** [comment]
@UnderGreen and @eldios can you please help me review this?

File: assets/marketplace/cloudformation/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    sh 'dd if=/dev/urandom bs=128 count=1 | tr -dc "a-z0-9" | fold -w 16 | head -n 1 > random-string'
+                    sh 'sed -i "s/jenkins-teleport-quickstart-test-stack.gravitational.io/jenkins-teleport-quickstart-$(cat random-string).gravitational.io/g" full-parameters.json'
+                    sh 'jq -r \'.[] | select(.ParameterKey == "DomainName") | .ParameterValue\' full-parameters.json > generated-domain-name'
+                    sh 'cat full-parameters.json'
+                    sh 'cat generated-domain-name'
+                    // update path to parameters file and create ASG stack
+                    sh 'JENKINS_CLOUDFORMATION_PARAMETERS_PATH=full-parameters.json make jenkins-create-stack'
+                    sh 'make jenkins-wait-for-stack-creation'
+                    sh 'make jenkins-get-s3-bucket-id > s3-bucket-id'
+                    sh 'cat s3-bucket-id'
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/Jenkinsfile]: Are 3 `cat` directives here for debugging here?

File: assets/marketplace/cloudformation/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    post {
+        always {
+            dir('assets/marketplace') {
+                script {
+                    // we want to run all these shutdown steps in order regardless of success/failure (in case part of the ASG fails)
+                    // so we use '|| true' to suppress return codes > 0
+                    sh 'make jenkins-delete-stack || true'
+                    sh 'make jenkins-wait-for-stack-deletion || true'
+                    sh 'make jenkins-delete-stack-vpc || true'
+                    sh 'make jenkins-wait-for-stack-deletion-vpc || true'
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/Jenkinsfile]: What if actually you get an error during deleting the stack and you skip deletion with that `|| true`.

File: assets/marketplace/cloudformation/Jenkinsfile-build-ent [hunk 3]
```diff
@@ -0,0 +1,27 @@
+#!groovy
+pipeline {
+    agent any
+    options {
+        ansiColor(colorMapName: 'XTerm')
+        disableConcurrentBuilds()
+        timestamps()
+    }
+    parameters {
+        string(name: 'version', defaultValue: '2.6.1', description: 'Teleport version to build')
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/Jenkinsfile-build-ent]: Can you set here `3.1.1` as in files above?

File: assets/marketplace/cloudformation/Jenkinsfile-build-oss [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#!groovy
+pipeline {
+    agent any
+    options {
+        ansiColor(colorMapName: 'XTerm')
+        disableConcurrentBuilds()
+        timestamps()
+    }
+    parameters {
+        string(name: 'version', defaultValue: '2.6.1', description: 'Teleport version to build')
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/Jenkinsfile-build-oss]: The same about version

File: assets/marketplace/cloudformation/Makefile [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+BUILD_SUBNET_ID ?=
+
+# Override for AMI name if needed
+BUILD_AMI_NAME ?=
+
+# Default build region
+AWS_REGION ?= us-west-2
+
+# Teleport version
+TELEPORT_VERSION ?= 3.0.1
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/Makefile]: 3.1.1?

File: assets/marketplace/cloudformation/Makefile [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+.PHONY: oss-name-build
+oss-name-build: TELEPORT_TYPE=oss
+oss-name-build:
+	@echo "Building image $(TELEPORT_VERSION) $(TELEPORT_TYPE) with name $(BUILD_AMI_NAME)"
+	packer build -force -var ami_name=$(BUILD_AMI_NAME) template.json
+
+.PHONY: ent
+ent: TELEPORT_TYPE=ent
+ent: check-ent-vars
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/Makefile]: Can you leave a comment here what is this target about? I am not sure I followed.

File: assets/marketplace/cloudformation/files/bin/teleport-generate-config [hunk 7]
```diff
@@ -0,0 +1,123 @@
+#!/bin/bash
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/files/bin/teleport-generate-config]: Can you set 
```
set -o nounset
set -o errexit
```
to be sure there is no unset variables and exit if error.

File: assets/marketplace/cloudformation/files/bin/teleport-get-cert [hunk 8]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# This script uses DNS-01 challenge, which means that users
+# have to control route53 zone as it modifes zone records
+# to prove to letsencrypt that they own the domain.
+
+set -e
+set -x
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/files/bin/teleport-get-cert]: Are you sure you need debugging output here?

File: assets/marketplace/cloudformation/files/bin/teleport-lock [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+set -x
+set -e
+
+# Source variables set up by cloudformation template
+. /etc/teleport.d/conf
+
+LOCK="/teleport/${TELEPORT_CLUSTER_NAME}"
+NOW=$(date +%s)
+TTL=$(($NOW+3660))
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/files/bin/teleport-lock]: 1 hour and 1 minute are intentional here?

File: assets/marketplace/cloudformation/files/bin/teleport-ssm-get-token [hunk 10]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# This is a service that fetches SSM token from parameter store
+# Note that in this scenario token is written to the file.
+# Script does not attempt to fetch token during boot, because the tokens are published after
+# Auth servers are started.
+
+set -e
+set -o pipefail
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/files/bin/teleport-ssm-get-token]: There are no pipes in the file to set this param.

File: assets/marketplace/cloudformation/files/install.sh [hunk 11]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+set -x
```
Comment (UnderGreen) [inline assets/marketplace/cloudformation/files/install.sh]: Debugging.

Comment (webvictim) [reply assets/marketplace/cloudformation/files/install.sh]: The debugging is useful for extra output if a build fails so I'll leave it in place.

File: assets/marketplace/Makefile [hunk 12]
```diff
@@ -26,192 +24,58 @@ INSTANCE_TYPE ?= t2.micro
...
 
-# Jenkins Cloudformation parameters path
-JENKINS_CLOUDFORMATION_PARAMETERS_PATH ?= ./files/build/jenkins-parameters.json
+# Telegraf version
+TELEGRAF_VERSION ?= 1.5.1
 
-# YAML filename
-CF_YAML_FILENAME ?= ./oss.yaml
+# InfluxDB version
+INFLUXDB_VERSION ?= 1.4.2
```
[the remark this answers is not in the document]
Comment (webvictim) [reply assets/marketplace/Makefile]: TIG stack is updated to the newest versions now.

**Review by UnderGreen — APPROVED** [review]

**Pull request merged**